wordpress-mobile/WordPress-Android
# [Tooling] Use a trusted CI agent for pipelines that require push access

### Description
This PR uses the trusted agent helper script `use-bot-for-git` to use the GitHub bot for performing git push operations on CI.

### Testing Steps
Since there's currently no active release in progress, I've already created a read-only deploy key and used it in our setup.

- Test run on 2597b0df0d9b28c2443bcb4155450c3f798450f0 (CI fails as expected with a read-only deploy key): https://buildkite.com/automattic/wordpress-android/builds/19176#019075ae-ce21-4c18-8ab4-aa5df80216de/120-188
- Test run on ab7a38e9168b3ad821e5b695efbd147c84956598, CI succeeded (commit 06711d04cb73eaec26e79b93c9b144c7fac139d9): https://buildkite.com/automattic/wordpress-android/builds/19178#019075b5-0fd5-4eea-b683-4dd3ddb83c86/127-181

This PR will be fully tested during the next release cycle of the project, when the affected pipelines should be able to finish successfully.

### Follow-up
Once this PR is merged, I will revoke the now unused deploy key.
